concepts/loops/about.md

```diff
@@ -0,0 +1,178 @@
+# About
+
+In Java there are four looping constructs, two that are condition centric: `while` and `do-while` and the other two are iteration centric: `for` and `for-each`.
+
+## For
+
+A for loop provides a mechanism to execute a group of statements repeatedly until some condition is met.
+
+```java
+for (initialization; test; update) {
+    body;
+}
+```
+
+The `initialization` sets an initial state for the loop and is executed exactly once at the start of the loop.
+Typically it declares and assigns a variable used in the test expression and update statement.
+For example:
+
+```java
+int i = 1
+```
+
+The `test` expression tests if the loop should end.
+If it evaluates to `true`, the body and then the update expression will be executed.
+If it evaluates to `false`, neither the body nor the update statement will be executed and execution resumes after the loop's closing bracket.
+Typically it checks the variable assigned in the initialization block.
+For example:
+
+```java
+i <= 10
+```
+
+After executing the loop body, the `update` expression is executed.
+Typically it increments or decrements the loop variable by some value.
+For example:
+
+```java
+i++
+```
+
+A `for` loop printing out the first four squares would look like this:
+
+```java
+for (int i = 1; i <= 4; i++) {
+    System.out.println("square of " + i + " is " + i * i);
+}
+```
+
+The output would be:
+
+```text
+square of 1 is 1
+square of 2 is 4
+square of 3 is 9
+square of 4 is 16
+```
+
+If iterating through every element in a collection, a `for-each` loop is preferred, but it can be done with a `for` loop like this:
```

Comment:
```suggestion
Iterating through every element in a collection is usually performed using a `for-each`, but it can be done with a `for` loop like this:
```